### DISCUSSION TITLE
January 2021 updates

### DISCUSSION THREAD

**Discussion by simprecicchiani** [simprecicchiani/ThinkPad-T460s-macOS-OpenCore#82] [General]
January 2021 updates are now live (https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/commit/95e328c2c1332f0630dfe3ca039290d49fe0eb0f). I didn't test the EFI yet and I won't release it until then (Edit: [released](https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/releases/tag/0.6.5.2)).
In the mean time you can download it directly by cloning the repo and test it.
Ocvalidate check was positive.
Trackpad drivers received a substantial update that should fix sleep problems like #64; I will try to enable `standby` and see what happens (Edit: standby is still broken).

**Comment by simprecicchiani** [comment]
From early testing it looks like `NVMeFix` ver. 1.0.5 causes startup issues. Reverted back to 1.0.4 in https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/commit/2c203204bf1e03dd43661b5349bf093123195f75.
`VoodooPS2` ver. 2.2.0 and `VoodooRMI` ver. 1.3 work as expected.

**Comment by simprecicchiani** [comment]
Some SSDTs missing from `/EFI/OC/Config.plist` caused incompatibility with YogaSMC. Fixed with https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/commit/e69812fb934e8df0f20164be3d41178d5c260dce.
[New release](https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/releases/tag/0.6.5.1) is updated with these changes.

**Comment by mhl221135** [comment]
RTCMemoryFixup.kext on latest commits 0.6.5 should be disabled as it is now ?


Also the "Quiet Balanced Performance" switch of yogasmc.app doesn't work, it just switches to Balanced if i try to switch it , and for what it is(i dont find description on github(= )?

**Comment by alexdelli** [comment]
I'm back to 6.4 and Simone EFI December, because I had a lot of issue after logon:
1) Lost the windows icon on boot
2) no verbose mode, only logon screen locked using wifi BSD so I witch to Airportltwm
3) after logon YogaSMC didn't works well and battery percentage not accurate ( I think is necessary to turn off SIP)

the point 1) is very very difficult to understand, because with EFI 6.4 all works great

**Reply by simprecicchiani** [reply]
> 1. Lost the windows icon on boot

In `/EFI/OC/Config.plist` under `Misc -> Security` try `ScanPolicy`:`0`

> 2. no verbose mode, only logon screen locked using wifi BSD so I witch to Airportltwm

Not sure what do you mean

> 3. after logon YogaSMC didn't works well and battery percentage not accurate (I think is necessary to turn off SIP)

I'll try so address the battery situation asap

**Comment by simprecicchiani** [comment]
I've just published a [new release](https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/releases/tag/0.6.5.2) for January 2021 updates.

**Reply by mhl221135** [reply]
There is no SSDT-BAT0.aml in ACPI folder, it should be BATT?

`<dict>
				<key>Comment</key>
				<string>Battery fixes for T460s</string>
				<key>Enabled</key>
				<true/>
				<key>Path</key>
				<string>SSDT-BAT0.aml</string>
			</dict>`

**Reply by mhl221135** [reply]
also which standby and hiberinate modes you use?

**Reply by simprecicchiani** [reply]
> There is no SSDT-BAT0.aml in ACPI folder, it should be BATT?

Correct, I forgot to change it

> also which standby and hiberinate modes you use?

hibernate mode 3 but unfortunately it's useless with standby disabled `0`

**Reply by simprecicchiani** [reply]
I did a full charge and discharge cycle and the batteries look ok to me. I can also confirm that when they discharge completely the laptop shuts down.
![Schermata 2021-01-17 alle 20 01 37](https://user-images.githubusercontent.com/46405079/104855425-05b30f80-590d-11eb-9130-a632ad35ac69.png)
![Schermata 2021-01-17 alle 20 49 04](https://user-images.githubusercontent.com/46405079/104855427-08156980-590d-11eb-9ff2-e9d09ccfbaf8.png)

**Reply by mhl221135** [reply]
> > There is no SSDT-BAT0.aml in ACPI folder, it should be BATT?
> 
> Correct, I forgot to change it
> 
Now its not showing percentage in tray, i can see it only in hwmonitorsmc
![Снимок экрана 2021-01-17 в 22 53 44](https://user-images.githubusercontent.com/55281358/104855690-db664f80-5916-11eb-87b6-7a1067823bda.png)

**Reply by alexdelli** [reply]
I'll try tomorrow with this new release, so I have to activate:
cpufriend ( with my profile )
airportlmw...
Nve fix?
other?

**Reply by mhl221135** [reply]
Did you tried to use laptop near 30minuts than keep it sleep than use again? as i seen its trouble happens after sleep cycle usually(now i am testing new relese but battery not showing in tray)
UPD(now it showing, after 2 minuts of system starts)

**Reply by mhl221135** [reply]
> I'll try tomorrow with this new release, so I have to activate:
> cpufriend ( with my profile )
> airportlmw...
> Nve fix?
> other?

Yes you have to activate 
NVME if your SSD is NVME 
airportitlwm or openbsd
and cpufriend just copy ssdt_data.aml from your old efi to EFI/OC/ACPI and replace SSDT-PLUG.aml by ssdt_data.aml in config.plist

Also you can find folowed sections with kext in files starting by #

and replace your SMBIOS ROM SERIAL etc.

![Снимок экрана 2021-01-17 в 22 58 30](https://user-images.githubusercontent.com/55281358/104855796-87a83600-5917-11eb-9b12-37bc7aa22bab.png)

**Reply by mhl221135** [reply]
> I did a full charge and discharge cycle and the batteries look ok to me. I can also confirm that when they discharge completely the laptop shuts down.
>
i test it, problem with power off after first battery discharged still exist(
maybe I can somehow write down the battery logs for you to solve this problem (

**Reply by simprecicchiani** [reply]
> i test it, problem with power off after first battery discharged still exist

This does not occur on my laptop. Could you check your full charge capacity with coconut battery?

**Reply by mhl221135** [reply]
> > i test it, problem with power off after first battery discharged still exist
> 
> This does not occur on my laptop. Could you check your full charge capacity with coconut battery?

coconut and hwmonitorsmc show near ~2300 it is sometimes changing from 2000-2500mAh

it is correct capacity becoyse both batterys weard by 50%

**Reply by soeryobadja** [reply]
> 
> Also you can find folowed sections with kext in files starting by #
> 

Hello.
I think I need some help with the # files here. Do I need to incorporate them? Or we need to choose which one to incorporate based on the devices we have? 
What about the SSDT with the #? Is it not required and for testing purpose only? 

Thanks.

**Reply by simprecicchiani** [reply]
Correct, merge the plist parts you need.

**Reply by soeryobadja** [reply]
Sorry I am still a bit lost. In which folder and in which part of config.plist should I put and define this plist files with "#"? AFAIK,  the only plist file is only config.plist?

What about the SSDT with "#"?

Thanks.

**Reply by simprecicchiani** [reply]
`#SSDT-***.aml` are SSDTs that have been replaced by other; don't mind them.
 
 `#***.plits` are optional parts for `Config.plist` (not picked up by OpenCore by themselves).

The idea is to keep the basic `Config.plist` as clean as possible given the multiple setup different people might need. If you need, for example, intel WiFi support, you open `#intel-wlan.plist` and merge its content into `Config.plist`.

In `#intel-wlan.plist` you'll find some kexts entry already in place (under `Kernel -> Add`) and it's just a matter of copy and paste them into the same section of `Config.plist`. This can be easily done with ProperTree.app by right-clicking on `Add` in `#intel-wlan.plist` and select `Copy children` then right-click on the same section `Add` of `Config.plist` and select `Paste`.

**Comment by alexdelli** [comment]
No, I have several errors, I don't know why.

Can I post my EFI if somebody can check it ? it's not urgent, when you have time

![1610967540588](https://user-images.githubusercontent.com/6674409/104907193-fbccf300-5984-11eb-924a-a023407c9d2c.jpg)
<img width="411" alt="Schermata 2021-01-18 alle 11 59 30" src="https://user-images.githubusercontent.com/6674409/104907206-012a3d80-5985-11eb-9628-c4c05e7146c8.png">
<img width="316" alt="Schermata 2021-01-18 alle 11 59 58" src="https://user-images.githubusercontent.com/6674409/104907213-038c9780-5985-11eb-8a59-41678c07d1c3.png">

**Reply by simprecicchiani** [reply]
You are using a 0.6.4 config.plist with OC 0.6.5. You have to remove the first entry under `UEFI -> Quirks` and use the proper SSDTs provided in release 0.6.5. It would be better if you can just use the new Config.plist and add your personal infos.

Off topic: Have fun with Serato DJ 😄

**Comment by alexdelli** [comment]
Now all works very well :)

**Comment by soeryobadja** [comment]
Hi all,

I am testing the 0.6.5 with Catalina 10.15.7 and everything seems to run well. Haven't check the battery, Think LED and using it for continuous work, though. 
Some question:
1. Are we no longer use SSDT-EC, as I saw it is under "#" mark?
2. For YogaSMC, how can I change the keys assignment? For example, I prefer F9 to open System Preferences, but it open YogaSMCPane instead. Also, The F10, I prefer it to open the spotlight and not Siri (as I don't use Siri). Any tips on how to do this? 

Thanks.

**Reply by simprecicchiani** [reply]
I tested the new battery patches (which involve `BATX` and `EC`) and they work fine on my machine. I've included them into [Release 0.6.6](https://github.com/simprecicchiani/ThinkPad-T460s-macOS-OpenCore/releases/tag/0.6.6).
I find your ideas about YogaSMC hotkeys implementation much more useful than default. It should be pretty easy to assign custom shortcut but haven't tried yet.

**Reply by soeryobadja** [reply]
I have not look into further details of YogaSMC yet, as I also still have limitation in understanding the technical details. 

From what I see, Spotlight in keyboard shortcut of system preference is already assigned by Cmd-Space, and I want to add F10 as spotlight just like the icon suggest. Same like F9 where it previously loads to system preference (not YogaSMCPane), which I assume it was implemented under YogaSMC.ketx unlike the previous implementation with Thinkpad apps via SSDT patch (which I can track the logic of this SSDT inside config.plist since I still have limited understanding). 

Will appreciate if someone has gone through this way using YogaSMC, even though it is not very urgent and important thing to be implemented for most people. 

I will look into the 0.6.6 version this weekend for further testing. 

Thanks.

**Reply by simprecicchiani** [reply]
There you go https://github.com/zhen-zen/YogaSMC/issues/109

**Reply by soeryobadja** [reply]
Yep. Saw that thread of yours as well. 

Question from a newbie, apologize for the simplistic approach:
1. Do I need to add the "optionFlag" as shown (was for IdeaEvents), or just modifying the `TP_HKEY_EV_SETTING.rawValue: [0: EventDesc("Settings", act: .prefpane, display: false)], // 0x101D` and change the `.prefpane` to `.systempreferences`? 
2. For changing F10 from Siri to Spotlight, should I just change `.siri` in `TP_HKEY_EV_SEARCH.rawValue: [0: EventDesc("Search", act: .siri, display: false)], // 0x101E` to `.spotlight`?

Thanks.

**Reply by simprecicchiani** [reply]
I’m sorry but I’m just as newbie as you are.
